modules/core/src/main/scala/lucuma/odb/api/model/syntax/LensOps.scala

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.api.model.syntax
+
+import cats.data.StateT
+import lucuma.odb.api.model.EitherInput
+import monocle.Lens
+
+final class LensOps[S, A](val self: Lens[S, A]) extends AnyVal {
+
+  def toState: StateT[EitherInput, S, A] =
+    StateT(s => Right((s, self.get(s))))
+
+  def st: StateT[EitherInput, S, A] =
+    toState
+
+  def extract: StateT[EitherInput, S, A] =
+    toState
+
+  def edit(a: A): StateT[EitherInput, S, A] =
+    assign(a)
+
+  @inline def :=(a: A): StateT[EitherInput, S, A] =
+    edit(a)
```

Comment:
It seemed better to tie down the effect in `StateT` than to add it to the `:=` method and ruin the syntax.  I don't know.  I guess it would get inferred.